framework/include/problems/SubProblem.h

```diff
@@ -1092,30 +991,94 @@ SubProblem::getFunctor(const std::string & name,
   _functor_to_requestors[name].insert(requestor_name);
 
   // Get the requested functor if we already have it
-  const auto & functors = _functors[tid];
-  auto it = functors.find(name);
-  if (it != functors.end())
+  auto & functors = _functors[tid];
+  auto find_ret = functors.find(name);
+  if (find_ret != functors.end())
   {
     if (functors.count(name) > 1)
       mooseError("Attempted to get a functor with the name '",
                  name,
                  "' but multiple functors match. Make sure that you do not have functor material "
                  "properties, functions, and variables with the same names");
 
-    const auto * const functor = dynamic_cast<const Moose::Functor<T> *>(it->second);
+    auto * const functor = dynamic_cast<Moose::Functor<T> *>(find_ret->second.get());
     if (!functor)
       mooseError("A call to SubProblem::getFunctor requested a functor named '",
                  name,
-                 "' that returns the type: ",
+                 "' that returns the type: '",
                  libMesh::demangle(typeid(T).name()),
```

Comment:
should we demangle?
Ive noticed it gives the metaphysicl long name for ADReal, but we would be happier with ADReal since this is a log users see

Reply:
should functors keep track of which object created them/ (or assigned them the last, if wrapper keeps the name)

Reply:
we should definitely demangle or else we get gobbly gook like `_ZN9wikipedia7article6formatEv`. Perhaps you're asking whether we should do "if string == long-metaphysicl-name, then string-replace with ADReal" and the answer is probably.

Reply:
> should functors keep track of which object created them/ (or assigned them the last, if wrapper keeps the name)

Probably at some point. Nothing I've done has been that complicated yet, but I could imagine a future where there are so many materials, etc. that you are unsure of where a function definition came from. That has certainly happened to me with regular materials

Reply:
Yeah that s the issue I m thinking of. FunctorFluidProps define so many for example, new users may not know where something came from. 

For the replace with ADReal, if you want to leave this for later, please let me know and I ll add an issue. This is something I think we need for user friendliness

Reply:
documented in #20076
#20077